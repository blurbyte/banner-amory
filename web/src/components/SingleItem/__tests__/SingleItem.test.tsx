import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { MockedProvider } from 'react-apollo/test-utils';
import * as delay from 'delay';

import { SingleItem, getSingleItem } from '../index';
import { testItem } from './mocks';

const mocks = [
  {
    request: {
      query: getSingleItem,
      variables: { slug: 'alettes-bracelet' }
    },
    result: {
      data: {
        item: {...testItem}
      }
    }
  }
];

beforeEach(cleanup);

test('renders loading state initially', () => {
  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <SingleItem slug="alettes-bracelet"/>
    </MockedProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('renders loaded item', async () => {
  const { container, getByTestId } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <SingleItem slug="alettes-bracelet"/>
    </MockedProvider>
  );

  await delay(0);

  expect(container.firstChild).toMatchSnapshot();
});
