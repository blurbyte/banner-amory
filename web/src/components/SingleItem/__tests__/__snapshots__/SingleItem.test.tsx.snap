// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loaded item 1`] = `
<article
  class="sc-htpNat eroWgC sc-bwzfXH ezMGVG"
>
  <h1
    class="sc-bxivhb godseO"
  />
  <div
    class="sc-ifAKCX eMgKLG"
  >
    <div
      class="sc-gZMcBi crAViM"
    >
      <img
        class="sc-EHOje dPoLsW"
        src="undefinedundefined.png"
      />
      <div
        class="sc-gqjmRU gcZhru"
      >
        <svg
          viewBox="0 0 50 90"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M50.005 90L25.003 67.452 0 90V0h50.005v90z"
            fill="currentColor"
          />
        </svg>
        <div
          class="sc-VigVT iEHYVi"
        />
      </div>
    </div>
    <div
      class="sc-jzJRlG eAtAcy"
    >
      <dl
        class="sc-bZQynM hthShY"
      >
        <dt
          class="sc-gzVnrw kiLclw"
        >
          Part
        </dt>
        <dd
          class="sc-htoDjs hoHbUn"
        >
          The Banner Saga undefined
        </dd>
      </dl>
      <dl
        class="sc-bZQynM hthShY"
      >
        <dt
          class="sc-gzVnrw kiLclw"
        >
          Rank
        </dt>
        <dd
          class="sc-htoDjs gBjpyW"
        />
      </dl>
      <dl
        class="sc-bZQynM hthShY"
      >
        <dt
          class="sc-gzVnrw kiLclw"
        >
          How to get
        </dt>
        <dd
          class="sc-htoDjs hoHbUn"
        >
          Unknown
        </dd>
      </dl>
    </div>
  </div>
</article>
`;

exports[`renders loading state initially 1`] = `
<article
  class="sc-htpNat eroWgC sc-bwzfXH ezMGVG"
/>
`;
